### PR TITLE
Restore ability to add instructors to assessment groups

### LINKED ISSUE #9297: Instructors cannot create a group with themselves in it
This can be considered a regression, possibly introduced by #9095. Instructors may want to join groups to test out group functionality, which is a valid use case. Creating a group as an instructor, whether that's via the "Add a group" button on the assessment groups page or via viewing the assessment as an instructor, fails with the following flash message:

> Failed to create the group XYZ. User user@example.com is not enrolled in this course

### PR DESCRIPTION
**Pull request by jonatanschroeder**
Resolves #9297. Also adds a test to ensure this is possible again in the future.

### CODE REVIEW

File: apps/prairielearn/src/tests/instructorAssessmentGroups.test.ts [hunk 1]
```diff
@@ -114,6 +114,26 @@ describe('Instructor group controls', () => {
...
+    const csrfToken = extractAndSaveCSRFToken({}, getResponse.$, '[name="add-group-form"]');
+
+    const response = await fetchCheerio(instructorAssessmentGroupsUrl, {
+      method: 'POST',
+      body: new URLSearchParams({
+        __csrf_token: csrfToken,
+        __action: 'add_group',
+        group_name: 'TestGroupWithInstructor',
+        // Add instructor to the group
+        uids: 'dev@illinois.edu',
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/tests/instructorAssessmentGroups.test.ts:128]: @nwalters512 is the hardcoded UID here ok or should I use the config parameter (or some other means) to get the instructor UID?

File: apps/prairielearn/src/lib/groups.ts [hunk 2]
```diff
@@ -229,21 +229,28 @@ export async function addUserToGroup({
...
-        })
-      : null;
-    if (enrollment == null) {
+        })) != null ||
+        (await sqldb.callRow(
+          'users_is_instructor_in_course_instance',
+          [user.user_id, group.course_instance_id],
+          z.boolean(),
+        ))
+      : false;
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/groups.ts:244]: Nit: I find this pretty hard to read. I'd say either a) lift this into a function where you can take advantage of early returns, or b) use use normal `if ...` to handle the branches.

File: apps/prairielearn/src/tests/instructorAssessmentGroups.test.ts (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/src/tests/instructorAssessmentGroups.test.ts:128]: Hardcoding is fine, I think.

**Review by nwalters512 — APPROVED** [review]

File: apps/prairielearn/src/lib/groups.ts (hunk 2 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/groups.ts]: I tried with normal `if` statements, but that made TS unable to identify that `user` is not null after this statement due to the multiple conditions. I have split the tests in 7c5492cd45869c452f54410a1ed6f69ee23b93c3, still taking advantage of short-circuit, but testing for instructor first.

**Pull request merged**